### PR TITLE
Issue #236: DemoExampleMaker: re-order steps functionality

### PR DESCRIPTION
**Pull request by Cthomp7**
**Finished the working version of the custom reorderable list!** 🎉

Reorderable list additions:
- Custom reorderable list created w/ styles
- Updated DEM to functionally work w/ new element
- Added a function to add IDs to future uploaded sketches w/out

Additional changes made:
- Removed `appearence: none;' under www/core/styles/base.css
[ This was causing the checkbox input element to not display ]
- Changed the DEM text input styles to fit the widget better

I am sure my code is not the most simplified, so let me know if there is anything I should change. I did run it through a JS standard checker and it looked good to go. (There were some flags on the DEM index.js, but not on things I added, so I left them alone) Additionally, I tried to troubleshoot this element to make sure everything was working. If you decide to merge this, in the future, if there is an issue feel free to ping me in the issue. :)

Hope you like it!! 🐸 #236

### CODE REVIEW

**Comment by Cthomp7** [comment]
Also! I made those new svg files myself, so they should be good to use.

**Comment by nbriz** [comment]
@Cthomp7 this is rad! i'm already using it to make demos for one of my classes (^__^) 

i'm noticing some issues though, i'll leave a comment here with all the bugs i find after i spend a bit more time using it.

@sarahrooney maybe u can give [the CSS](https://github.com/netizenorg/netnet.studio/pull/245/files#diff-c516e73fa624ba5c9c03dc0445a1bc3064e51fbadb40911d41597d5c3cfe7baa) a look over && make sure it conforms to the BEM/CSS Style Guide (which we probably need to document at some point)

File: www/core/styles/base.css [hunk 1]
```diff
@@ -47,7 +47,6 @@ body {
 
 button,
 input {
-  appearance: none;
   outline: none;
 }
```
Comment (nbriz) [inline www/core/styles/base.css:51]: @sarahrooney do u remember what this was for? is it ok to remove this.

**Comment by Cthomp7** [comment]
@nbriz I believe I fixed all the bugs we encountered.
- updating reorderable list in DEM with a code example loaded on page (both `ex=...` and `#example/`)
- reordering steps bug (this fixed the arrow functionality too)
Side note: my branch was really messy for some reason so my commits are mushed with pulling my original commit and resolving any conflicts. If you have any questions about that let me know.

**Comment by nbriz** [comment]
awesome @Cthomp7 ! seems like most the issues we talked about are fixed! 
except i'm still getting the "null" titles showing up on steps that get re-ordered (see attached video)

https://github.com/netizenorg/netnet.studio/assets/2506806/34305638-3905-45e8-b48a-334883761043

**Comment by Cthomp7** [comment]
@nbriz This should be working now. Have the prettier code extension for work so it added a bunch of semicolons and additional non-standard JS formatting, so I had to go back and fix that. Let me know if you have any issues and I will try to knock them out quick today/tomorrow!

**Comment by nbriz** [comment]
@Cthomp7 looks great! gonna merge it!

**Pull request merged**